File: ftw_backend/src/models/type_overview.py
from dataclasses import dataclass, field
from sqlalchemy.orm import Session  # type: ignore
from .category_type import CategoryType
from database.schemas import CategorySchema, TransactionSchema
from typing import Any

@dataclass
class CategorySummary:
    category_name: str = ""
    category_amount: int = 0

@dataclass
class MonthOverview:
    category_type: CategoryType = ""
    type_overview: list[CategorySummary] = field(default_factory=list)

@dataclass
class YearOverview:
    year_overview: list[dict[str, Any]] = field(default_factory=list)

File: ftw_frontend/src/assets/types/TypeOverview.ts
export type CategorySummary = {
  category_name: string;
  category_amount: number;
};

export type MonthOverview = {
  type_name: string;
  type_overview: CategorySummary[];
};

export type YearOverview = {
  year_overview: Map<string, any>[];
};
